py-nvitop: new package

nboelte opened the pull request:
Adding nvitop from https://github.com/XuehaiPan/nvitop

nboelte committed:
Adding nvitop package

nboelte committed:
[@spackbot] updating style on behalf of nboelte

tldahlgren commented on var/spack/repos/builtin/packages/py-nvitop/package.py:
```diff
@@ -0,0 +1,25 @@
+    license("Apache-2.0")
+
+    version("1.4.0", sha256="92f313e9bd89fe1a9d54054e92f490f34331f1b7847a89ddaffd6a7fde1437bb")
+
```
According to `pyproject.toml`, the minimum `python` is `3.7` so you could add a dependency on that version. (Note there's only `3.6` left in Spack's package and it is deprecated.)

nboelte replied:
Thanks, i added python@3.7: as a dependency

tldahlgren commented on var/spack/repos/builtin/packages/py-nvitop/package.py:
```diff
@@ -0,0 +1,25 @@
+
+    version("1.4.0", sha256="92f313e9bd89fe1a9d54054e92f490f34331f1b7847a89ddaffd6a7fde1437bb")
+
+    depends_on("py-nvidia-ml-py")
+    depends_on("py-psutil")
+    depends_on("py-cachetools")
+    depends_on("py-termcolor")
```
You'll want to specify the dependency types (see https://spack.readthedocs.io/en/latest/packaging_guide.html#dependency-types) and constrain the versions of the dependencies (see https://spack.readthedocs.io/en/latest/packaging_guide.html#specifying-backward-and-forward-compatibility) based on the `dependencies` information in `pyproject.toml`:

```
dependencies = [
    # Sync with nvitop/version.py and requirements.txt
    "nvidia-ml-py >= 11.450.51, < 12.561.0a0",
    "psutil >= 5.6.6",
    "cachetools >= 1.0.1",
    "termcolor >= 1.0.0",
    "colorama >= 0.4.0; platform_system == 'Windows'",
    "windows-curses >= 2.2.0; platform_system == 'Windows'",
]
```

or `nvitop.egg-info/requires.txt`:

```
nvidia-ml-py<12.561.0a0,>=11.450.51
psutil>=5.6.6
cachetools>=1.0.1
termcolor>=1.0.0

[:platform_system == "Windows"]
colorama>=0.4.0
windows-curses>=2.2.0

[cuda10] 
nvidia-ml-py==11.450.51
```

such as:

```
    depends_on("py-colorama@0.4:", when="platform=windows", type=("build", "run"))
```

If you don't want to deal with support for windows, then add a conflict (see https://spack.readthedocs.io/en/latest/packaging_guide.html#conflicts-and-requirements).

nboelte replied:
Hi, I added the dependency types and version constraints, i was not really sure how to distinguish a build/run type dependency in this case so i set everything to run except for setuptools.

tldahlgren reviewed (changes requested):
Confirmed the homepage and version sha256 but see comments regarding dependencies.

Since you're adding the package, would you be willing to be a maintainer (see https://spack.readthedocs.io/en/latest/packaging_guide.html#maintainers) for it?

nboelte committed:
Adding dependency constraints and types

nboelte committed:
Adding maintainer

nboelte committed:
Adding python version dependency

nboelte committed:
Merge branch 'packages/py-nvitop' of github.com:nboelte/spack into packages/py-nvitop

nboelte commented:
I added myself as a maintainer and i added a conflict for platform=windows because it would require adding `py-windows-curses` which i definitely don't want to deal with.

nboelte committed:
Style fix

tldahlgren commented on var/spack/repos/builtin/packages/py-nvitop/package.py:
```diff
@@ -0,0 +1,33 @@
+    depends_on("py-termcolor@1.0.0:", type=("build", "run"))
+    depends_on("python@3.7:", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+    depends_on("py-colorama@0.4:", when="platform=windows", type=("build", "run"))
```
Do you want to keep this after adding the conflict at line 33?

nboelte replied:
Yeah that line could actually be confusing. I have commented it out as well and reworded the comment, thanks!

nboelte committed:
Commented out unused platform=windows dependency

nboelte committed:
Merge branch 'spack:develop' into packages/py-nvitop
